packages/frontend/src/context/User.ts

```diff
@@ -271,16 +271,19 @@ export class User {
 
     async getAirdrop() {
         if (!this.id || !this.userState) throw new Error('Identity not loaded')
+        if (this.reputation >= 0) throw new Error('do not need to airdrop')
+
         await this.unirepConfig.loadingPromise
         const unirepSocial = new ethers.Contract(
             this.unirepConfig.unirepSocialAddress,
             config.UNIREP_SOCIAL_ABI,
             config.DEFAULT_ETH_PROVIDER
         )
         // generate an airdrop proof
-        const attesterId = this.unirepConfig.attesterId
-        const { proof, publicSignals } =
-            await this.userState.genUserSignUpProof(BigInt(attesterId))
+        const negRepProof = await this.userState.genNegativeRepProof(
+            BigInt(this.unirepConfig.attesterId),
+            BigInt(Math.abs(this.reputation))
```

Comment:
I think it can be `min(Math.abs(this.reputation), this.subsidyReputation)`
so that it won't reveal the exact `this.reputation` through api if `Math.abs(this.reputation)>this.subsidyReputation`

Reply:
I finally choose to put `defaultAirdropKarma` to `UnirepConfig` since I found that in the backend, we also have a constant `DEFAULT_AIRDROPPED_KARMA`. Maybe in the future, we should put this in unirep social contract?

Reply:
actually the meaning of `defaultAirdropKarma` is not the airdrop subsidy
it means the previous airdrop system (user sign up with this airdrop karma)
Yeah I think the `subsidy`, (should be queried by 
```ts
const subsidy = (await this.unirepConfig.unirepSocial.subsidy()).toNumber()
```
)
can be put in `UnirepConfig`
and it is already in unirep social contract

so with current unirep social smart contract you can find
```ts
airdroppedReputation = 0
subsidy = 30 // used in airdrop subsidy
```